### PR TITLE
fix #173381, part 1: handle selection after MMRest reconstruction

### PR DESCRIPTION
**Pull request by dmitrio95**
This pull request is intended to fix crashes which happen if one follows scenarios described in these two issue comments ([first](https://musescore.org/en/node/173381#comment-852459), [second](https://musescore.org/en/node/173381#comment-856760)). These crashes are fixed by handling selection after multi-measure rests are reconstructed and, thus, after changes in the segments' structure of the score. Still this alone is not enough to fix the issue so there are some not very nice hacks in the solution which still make MuseScore have a predictable behavior in the given scenarios.

Still I recorded the described issues in scripts which are added to MuseScore tests which should help us avoid repeating these bugs later.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Travis CI failed, see https://travis-ci.org/musescore/MuseScore/jobs/454452854#L5621-L5660

Could you use the same diff options for those tests are we're using in the other mtests?

**Comment by dmitrio95** [comment]
It seems something in the last commits changes selection behavior a bit in a strange way. Anyway test scripts should not depend on it so I should make it possible to describe selection in some more precise way. As a temporary solution I just altered the test script for this case a bit not altering the resulting score.

Concerning diff options I certainly could do that but not right away. The problem is that I had to build that script framework inside the main `mscore` executable where test dependencies should not be normally available. Still the desired effect can be achieved if I make that engine save resulting scores instead of performing comparison by itself. I will think how is it better to be done in order not to break the other things working well about the new tests.

File: libmscore/select.h [hunk 1]
```diff
@@ -130,6 +130,9 @@ class Selection {
       Segment* _startSegment;
       Segment* _endSegment;         // next segment after selection
 
+      int _plannedTick1 = -1; // will be actually selected on updateSelectedElements() call
```
Comment (anatoly-os) [inline libmscore/select.h]: `_plannedTick`? Absolutely not clear what this variable does, unfortunately. I suppose it somehow defines begin/end range ticks to be selected after some operation, doesn't it? But better name or explaining comment will help to get the meaning better.

File: libmscore/select.cpp [hunk 2]
```diff
@@ -450,6 +450,20 @@ void Selection::appendChord(Chord* chord)
...
 void Selection::updateSelectedElements()
       {
+      if (_state == SelState::RANGE && _plannedTick1 != -1 && _plannedTick2 != -1) {
+            const int staffStart = _staffStart;
+            const int staffEnd = _staffEnd;
+            deselectAll();
+            Segment* s1 = _score->tick2segmentMM(_plannedTick1);
+            Segment* s2 = _score->tick2segmentMM(_plannedTick2, /* first */ true);
+            if (s2 && s2->measure()->isMMRest())
+                  s2 = s2->prev1MM(); // HACK both this and the previous "true"
```
Comment (anatoly-os) [inline libmscore/select.cpp]: It is better to explain the workaround in particular to leave code clear for other developers and for yourself in a year :)

Comment (dmitrio95) [reply libmscore/select.cpp]: I updated comments for both discussed places.

**Pull request merged**